pkg/hooks/exec/exec.go

```diff
@@ -18,11 +18,12 @@ const DefaultPostKillTimeout = time.Duration(10) * time.Second
 
 // Run executes the hook and waits for it to complete or for the
 // context or hook-specified timeout to expire.
-func Run(ctx context.Context, hook *rspec.Hook, state []byte, stdout io.Writer, stderr io.Writer, postKillTimeout time.Duration) (hookErr, err error) {
+func Run(ctx context.Context, hook *rspec.Hook, cwd string, state []byte, stdout io.Writer, stderr io.Writer, postKillTimeout time.Duration) (hookErr, err error) {
```

Comment:
Breaking the API is always a bit problematic as we need to update all users at the same time.
I see you have PRs for podman/buildah but I think cri-o is using this as well.

Also at this point the list of parameters is quite long (too long for me). It would make more sense to add a struct with the options and use it as parameter. That way we can add further fields in the future without breaking the API.

Reply:
Should we create a new API with an option?  RunH?

Reply:
@Luap99 I just checked out the cri-o repo and cannot find a direct use of the hook exec functions 

@vrothberg 
Yeah, I think it makes sense to pass in a struct instead. I have updated this PR to take an option struct instead.

@rhatdan 
Also, yeah, I see it could be difficult for the downstream projects if we introduce breaking change to the API here. So I created a new alt PR. Instead of changing the old functions, I created new functions with option struct and deprecated the old ones.

Please have a look here:
https://github.com/containers/common/pull/1521